workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/internal/flows/confidential/AnonymisePartiesFlow.kt

```diff
@@ -7,16 +7,20 @@ import net.corda.core.identity.AbstractParty
 import net.corda.core.identity.AnonymousParty
 import net.corda.core.identity.Party
 
-// TODO this should be internal?
+/**
+ * This flow notifies prospective token holders that they must generate a new key pair. As this is an in-line sub-flow,
+ * we must pass it a list of sessions, which _may_ contain sessions for observers. As such, only the parties that need
+ * to generate a new key are sent a [ActionRequest.CREATE_NEW_KEY] notification and everyone else is sent
+ * [ActionRequest.DO_NOTHING].
+ */
 class AnonymisePartiesFlow(
         val parties: List<AbstractParty>,
         val sessions: List<FlowSession>
 ) : FlowLogic<Map<Party, AnonymousParty>>() {
     @Suspendable
     override fun call(): Map<Party, AnonymousParty> {
-        // TODO: Need more checking here - that list of session parties is equal to list of parties supplied.
-        // TODO: This code will require updating to support the use of accounts.
-        // for now, it is assumed that confidential identities will represent legal identities, only.
+        val sessionParties = sessions.map(FlowSession::counterparty)
+        require(sessionParties.containsAll(parties)) { "You must provide sessions for all parties." }
```

Comment:
Is it correct? `sessionParties` are well known identities, but parties is a list of `AbstractParty`, so you can pass anonymous parties

Reply:
hmm yes it's a bit confusing but these abstract parties must actually be `Party` otherwise we wouldnt be able to initiate a session!

Reply:
I figured they are all `Party`s as we do this: 

```
        return sessions.mapNotNull { session ->
            val party = session.counterparty
            if (party in parties) {
```

Reply:
I'll change the type of `parties` to `List<Party>`